app/controllers/api/v1/journeys_controller.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class JourneysController < ApiController
+      before_action :validate_params, only: %i[create update]
+      after_action :create_event, only: %i[create update]
+
+      PERMITTED_NEW_JOURNEY_PARAMS = [
+          :type,
+          attributes: [:timestamp, :billable, vehicle: {}],
+          relationships: [from_location: {}, to_location: {}],
+      ].freeze
+
+      PERMITTED_UPDATE_JOURNEY_PARAMS = [
+          :type,
+          attributes: [:timestamp, :billable, vehicle: {}],
+      ].freeze
+
+      def index
+        paginate journeys
+      end
+
+      def show
+        render json: journey, status: :ok
+      end
+
+      def create
+        authorize!(:create, journey)
+        journey.save!
+        render json: journey, status: :created
+      end
+
+      def update
+        journey.update!(update_journey_attributes)
+        render json: journey, status: :ok
+      end
+
+    private
+
+      def move
+        @move ||= Move.accessible_by(current_ability).find(params.require(:move_id))
+      end
+
+      def journeys
+        @journeys ||= move.journeys.accessible_by(current_ability).default_order
+      end
+
+      def journey
+        @journey ||= if action_name == 'create'
```

Comment:
I think this would be nicer as two separate and explicit methods:

```ruby
def new_journey
  @new_journey ||= Journey.new(new_journey_attributes)
end

def journey
  @journey ||= Journeys.find(params.require(:id))
end
```

And then use the respective version above (just need to reference `new_journey` in create method, the other endpoints don't need changing):

```ruby
def create
  authorize!(:create, new_journey) # As mentioned above, I think this line is redundant anyway
  new_journey.save!
  render json: new_journey, status: :created
end
```

Reply:
@smoothcontract I did have something similar in an earlier version... but the `if` statement doesn't go away  - see the `create_event` method at the bottom of the controller. That needs to reference the `journey` and so will need to determine whether its `journey` or `new_journey`. On balance I decided always calling the journey `journey` simplified things.

Reply:
Ah, I didn't spot that extra use of this method. There is a work around you could use, although it might make Rubocop grumpy. If you memoized `new_journey` to the same `@journey` attribute then when `journey` is called later to build the event it would still add the event for the newly created journey.

```ruby
def new_journey
  @journey ||= Journey.new(new_journey_attributes)
end
```

That avoids the branching code, but it is a bit of a side effect I guess.

Reply:
hmm... nope, I don't like `@journey` in new_journey as its not very clear what's going on. if its all the same, I think I'd prefer just to call the journey `journey` and have the if statement

Reply:
Cool 👍